safenode/src/storage/chunks.rs

```diff
@@ -6,181 +6,82 @@
+use clru::CLruCache;
+use std::{num::NonZeroUsize, sync::Arc};
+use tokio::sync::RwLock;
+use tracing::trace;
 
-/// The XorName of the provided `Chunk`
-#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize, Debug)]
-pub struct ChunkAddress(pub XorName);
-impl ChunkAddress {
-    /// Returns the name.
-    pub fn name(&self) -> &XorName {
-        &self.0
-    }
-}
+const CHUNKS_CACHE_SIZE: usize = 20 * 1024 * 1024;
 
 /// Operations on data chunks.
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub(super) struct ChunkStorage {
-    file_store_path: PathBuf,
-}
-
-/// Chunk, an immutable chunk of data
-#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, custom_debug::Debug)]
-pub struct Chunk {
-    /// Network address. Omitted when serialising and
-    /// calculated from the `value` when deserialising.
-    address: ChunkAddress,
-    /// Contained data.
-    #[debug(skip)]
-    value: Bytes,
+    cache: Arc<RwLock<CLruCache<ChunkAddress, Chunk>>>,
 }
 
-impl Chunk {
-    /// Creates a new instance of `Chunk`.
-    pub fn new(value: Bytes) -> Self {
+impl Default for ChunkStorage {
+    fn default() -> Self {
+        let capacity =
+            NonZeroUsize::new(CHUNKS_CACHE_SIZE).expect("Failed to create in-memory Chunk storage");
         Self {
-            address: ChunkAddress(XorName::from_content(value.as_ref())),
-            value,
+            cache: Arc::new(RwLock::new(CLruCache::new(capacity))),
         }
     }
-
-    /// Returns the value.
-    pub fn value(&self) -> &Bytes {
-        &self.value
-    }
-
-    /// Returns the address.
-    pub fn address(&self) -> &ChunkAddress {
-        &self.address
-    }
-
-    /// Returns the name.
-    pub fn name(&self) -> &XorName {
-        self.address.name()
-    }
-
-    /// Returns size of contained value.
-    pub fn payload_size(&self) -> usize {
-        self.value.len()
-    }
-
-    /// Returns size of this chunk after serialisation.
-    pub fn serialised_size(&self) -> usize {
-        self.value.len()
-    }
-}
-
-impl Serialize for Chunk {
-    fn serialize<S: Serializer>(&self, serialiser: S) -> Result<S::Ok, S::Error> {
-        // Address is omitted since it's derived from value
-        self.value.serialize(serialiser)
-    }
-}
-
-impl<'de> Deserialize<'de> for Chunk {
-    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        let value = Deserialize::deserialize(deserializer)?;
-        Ok(Self::new(value))
-    }
 }
 
 impl ChunkStorage {
-    /// Creates a new `ChunkStorage` at the specified root location
-    ///
-    /// If the location specified already contains a `ChunkStorage`, it is simply used
-    ///
-    /// Used space of the dir is tracked
-    pub(super) fn new(path: &Path) -> Self {
-        Self {
-            file_store_path: path.join(CHUNKS_STORE_DIR_NAME),
-        }
+    #[allow(dead_code)]
+    pub(super) async fn addrs(&self) -> Vec<ChunkAddress> {
```

Comment:
I know this is since old. But maybe we can write out the term to use properly here?